### DISCUSSION TITLE
SOC Alerts view freeze using groupby

### DISCUSSION THREAD

**Discussion by 0nnyx** [Security-Onion-Solutions/securityonion#5764] [Unsupported Versions] [answered]
Using SO 2.3.80 and previous, groupby tends not to work and freeze the view.
I could reproduce this behavior systematically when using Playbook alerts:
1. Drilldown on an alert with several occurrences
2. Taking the example of playbook alerts for Autorun Keys Change, click groupby event_data.process.executable
3. The view doesn't change to show the column process.executable and gets frozen :  alert items cannot be expanded anymore

Same symptoms appear with Suricata alerts depending on the chosen grouping : view not freezing when grouping on GeoIP infos but freezing for example when clicking groupy network.community_id

Workaround: Use Kibana and toggle the 'add to view' for the column

Any similar report ? If not, please check this behavior.

**Comment by dougburks** [comment]
I tried grouping Suricata alerts by network.community_id and was unable to duplicate this issue using Google Chrome. What browser are you using?

**Reply by 0nnyx** [reply]
I'm using latest Chrome 94.0.4606.71 on Win10. Just tried again with some alerts and still same issue. I don't have the problem when grouping by rule.uuid (switching view to display only Count and rule.uuid)
Example: rule.name: "ET INFO Observed DNS Query to .biz TLD" Group: network.community_id
The view doesn't change after applying the groupby (still showing Timestamp, rule.name, event.severity_label, ...) and each alert cannot be expanded/collapsed anymore using the > sign left to the acknowledge bell icon.

**Comment by jertel** [comment]
There's some confusion with this discussion. You've mentioned being unable to expand grouped alerts (See bullet #3 in your original post and then again you mention this in the immediate post above.) Yet grouped alerts are not supposed to have an expand icon. When you mention 'freezing' I think of a browser that no longer responds to any keyboard or mouse click and must be force killed. Is this what you're seeing?

A video grab showing the problem would be useful here.

**Reply by 0nnyx** [reply]
I understand the confusion with the word "freeze". My browser doesn't freeze at all. I meant that the view doesn't change at all after clicking groupby.
So to rephrase, when using certain groupby items like network.community_id with Suricata alerts, the Alerts page doesn't change to display only Count and network.community_id. Instead The Alerts page remains as it was and clicking the expand/collapse icon doesn't have any effect anymore. Trying then another groupby will also have no effect (looking like the Alert page and not the browser is in frozen state). The only way to expand alerts is to go back to the main Alerts page. Attaching a screenshot showing it. Let me know if this is clear enough
![Capture](https://user-images.githubusercontent.com/54419678/136209400-97bcc087-7d82-42d8-8ecd-97c7bd99d60f.JPG)

**Comment by 0nnyx** [comment, marked as answer]
You can close this QA. Trying with Edge browser, I don't have that problem with the groupby. Probably linked to Chrome interpretation or caching. Might still help anyone who would experience similar issues.

**Reply by 0nnyx** [reply]
For sake of being complete, the cause was browser cache. Clearing Chrome cache allowed proper groupby.

**Reply by chergett** [reply]
Clearing the cache does not fix this same issue for me.  I have the issue in Chrome, Edge, and Firefox.  It worked a few times in Edge and then stopped working.  I've cleared the cache in all three browsers but the grouping still does exactly what was described above.  I'm on SOS version 2.3.91.

**Reply by chergett** [reply]
Did some further testing on this and it appears to (at least for me) be related to a cookie or cookies.  I'm not sure.  I cleared both cache and cookies, logged back in, and the group by for alerts was working.  If I try to sort alerts by severity, that breaks it.  I left the default sorting in place and so far, it's working again.

**Reply by dougburks** [reply]
@chergett You're replying to an old discussion from many months ago that has been marked as answered. From https://github.com/Security-Onion-Solutions/securityonion/discussions/1720:
> Start a new discussion instead of replying to somebody else's discussion
Please search to see if you can find similar discussions that may help you. However, in order to avoid confusion, please do NOT reply to somebody else's discussion with your own issue. Instead, please start a new discussion and in that new discussion you can provide a hyperlink to the related discussion.
